### PR TITLE
fix:  get comments

### PR DESCRIPTION
**Pull request by wowq**
Fixed the issue of only getting the first page of comments.

### CODE REVIEW

**Comment by ankitm123** [comment]
Can you add a test case for this?

**Comment by wowq** [comment]
Sorry, I also want to add some test cases, but I have referred to the test cases in the project, but I still can't find out how to add them.

File: pkg/cmd/pr/variables/variables.go [hunk 1]
```diff
@@ -175,16 +175,28 @@ func (o *Options) modifyVariables(text string) error {
...
 
 func (o *Options) loadPRComments(envVars map[string]string, pr *scm.PullRequest) error {
+	var res *scm.Response
+	var comments []*scm.Comment
+	var allComments []*scm.Comment
+	var err error
+
 	ctx := o.GetContext()
 	prNumber := pr.Number
+	firstRun := true
```
Comment (ankitm123) [inline pkg/cmd/pr/variables/variables.go:185]: would be good to add a comment explaining what is `firstRun`.

**Comment by msvticket** [comment]
@dependabot rebase
/ok-to-test
/approve

**Pull request merged**